app/src/main/java/ch/blinkenlights/android/vanilla/CoverCache.java

```diff
@@ -463,6 +468,19 @@ public Bitmap createBitmap(Song song, long maxPxCount) {
 					}
 				}
 
+				if (inputStream == null && (CoverCache.mCoverLoadMode & CoverCache.COVER_MODE_INLINE) != 0)
+				{
+					MediaMetadataRetriever mmr = new MediaMetadataRetriever();
+					mmr.setDataSource(song.path);
+
+					byte[] data = mmr.getEmbeddedPicture();
+					if (data != null) {
+						sampleInputStream = new ByteArrayInputStream(data);
```

Comment:
That looks like an error, looks like this wouldn't even compile?

Reply:
compiled and debugged, it delivers the desired result